lib/shared/bucketing-assembly-script/assembly/types/configBody.ts

```diff
@@ -111,10 +129,19 @@ export class ConfigBody {
                 this.audiences.set(audience_id, new NoIdAudience(aud as JSON.Obj))
             }
         }
-        const variables = getJSONArrayFromJSON(configJSONObj, 'variables')
-        this.variables = variables.valueOf().map<Variable>((variable) => {
-            return new Variable(variable as JSON.Obj)
-        })
+        const variablesJSON = getJSONArrayFromJSON(configJSONObj, 'variables').valueOf()
+        const variables = new Array<Variable>()
+        const _variableKeyMap = new Map<string, Variable>()
+        const _variableIdMap = new Map<string, Variable>()
+        for (let i=0; i < variablesJSON.length; i++) {
+            const variable = new Variable(variablesJSON[i] as JSON.Obj)
+            variables.push(variable)
+            _variableKeyMap.set(variable.key, variable)
+            _variableIdMap.set(variable._id, variable)
```

Comment:
do we actually need the `_variableIdMap` at the config level? I only see the `_variableKeyMap` being used.

If we do need it, I think it would be slightly more space efficient (and practically the same time efficiency) to have a `_variableIdKey` map that holds the _id to key mapping. That way we dont need to store every variable twice at the config level.

Reply:
in theory the two should be equivalent because its not storing the variable in each one, its just storing a reference to it

Reply:
yea technically we need both, `_variableIdMap` is used here: https://github.com/DevCycleHQ/js-sdks/pull/414/files#diff-e3b505010bb193bc1aedac8338f00363231ff4b24b565eb0989f678b2e772962R279

and the `_variableKeyMap` is used here: https://github.com/DevCycleHQ/js-sdks/pull/414/files#diff-e3b505010bb193bc1aedac8338f00363231ff4b24b565eb0989f678b2e772962R325